invoke.go

```diff
@@ -124,13 +124,14 @@ func findMissingDependencies(c containerStore, params ...param) []paramSingle {
 	for _, param := range params {
 		switch p := param.(type) {
 		case paramSingle:
-			if ns := c.getAllValueProviders(p.Name, p.Type); len(ns) == 0 && !p.Optional {
+			allProviders := c.getAllValueProviders(p.Name, p.Type)
+			_, hasDecoratedValue := c.getDecoratedValue(p.Name, p.Type)
+			if len(allProviders) == 0 && !hasDecoratedValue && !p.Optional {
```

Comment:
i had to pause and think about why we want to exempt list of deps if they are decorated. can u maybe explain in code and here about the special case.